lib/magicbell-rails.rb

```diff
@@ -11,5 +11,22 @@ module Rails
     def self.bell(args)
       Notification.bell(args)
     end
+
+    def self.fetch_categories(external_id)
+      response = client.user_with_external_id(external_id).notification_preferences.retrieve.attributes
+
+      response['categories']&.map { |category| category['slug'] } || []
+    end
```

Comment:
As new we building a library, we should make it a bit more flexible. So thinking we should return category objects.

then it could be used like this in our code for example:

```ruby
category_slugs = Magicbell::Rails.fetch_categories(external_id).map(&:slug)
```

this way if we ever need any other info of the category in future we could get it.

Also it would be good to keep the top level api methods very simple and push specifics into another class.

```suggestion
    sig { returns(T::Array[UserCategory]) } # we don't have sorbet on this gem but just trying to show you what i think it could return
    def self.fetch_categories(external_id)
      UserCategory.fetch(external_id)
    end
```